main/src/library/Fixpoint3/Boxable.flix

```diff
@@ -23,18 +23,21 @@ mod Fixpoint3.Boxable {
     /// Boxes the given `x`.
     ///
     @Internal
-    pub def box(x: a): Boxed with Order[a] = typematch x {
-        case v: Bool    => Boxed.BoxedBool(v)
-        case v: Char    => Boxed.BoxedChar(v)
-        case v: Int8    => Boxed.BoxedInt8(v)
-        case v: Int16   => Boxed.BoxedInt16(v)
-        case v: Int32   => Boxed.BoxedInt32(v)
-        case v: Int64   => Boxed.BoxedInt64(v)
-        case v: Float32 => Boxed.BoxedFloat32(v)
-        case v: Float64 => Boxed.BoxedFloat64(v)
-        case _: _ =>
-            let cmp = (o1, o2) -> Order.compare(unchecked_cast(o1 as a), unchecked_cast(o2 as a));
-            Boxed.BoxedObject(unchecked_cast(x as Object), cmp)
+    pub def box(x: a): Boxed with Order[a] = {
+        use Reflect.JvmValue;
+        match Reflect.reflect(x) {
+            case JvmValue.JvmBool(v)  => Boxed.BoxedBool(v)
+            case JvmValue.JvmChar(v)  => Boxed.BoxedChar(v)
+            case JvmValue.JvmInt8(v)  => Boxed.BoxedInt8(v)
+            case JvmValue.JvmInt16(v) => Boxed.BoxedInt16(v)
+            case JvmValue.JvmInt32(v) => Boxed.BoxedInt32(v)
+            case JvmValue.JvmInt64(v) => Boxed.BoxedInt64(v)
+            case JvmValue.JvmFloat32(v) => Boxed.BoxedFloat32(v)
+            case JvmValue.JvmFloat64(v) => Boxed.BoxedFloat64(v)
+            case JvmValue.JvmObject(v) =>
+                let cmp = (o1, o2) -> Order.compare(unchecked_cast(o1 as a), unchecked_cast(o2 as a));
```

Comment:
This is a really interesting case of combining traits and jvm representation. This might be a problem for enum optimization but not for the mission to remove typematch